### PR TITLE
Added analog pH sensor support

### PR DESCRIPTION
**Pull request by gorootde**
## Description:

Introduces support for pH Sensors with analog output, connected to an analog input of tasmota.

Supports calibration of probe by using 2 calibration solutions with low (A) and high (B) pH:
`AdcParam 9,<pH of solution A>,<ADC reading for solution A>,<pH of solution A>,<ADC reading for solution B>`

## Checklist:
  - [x] The pull request is done against the latest development branch
  - [x] Only relevant files were touched
  - [x] Only one feature/fix was added per PR and the code change compiles without warnings
  - [x] The code change is tested and works on Tasmota core ESP8266 V.2.7.4.9
  - [ ] The code change is tested and works on Tasmota core ESP32 V.1.0.5-rc4
    - I don't own an ESP32
  - [x] I accept the [CLA](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md#contributor-license-agreement-cla).

_NOTE: The code change must pass CI tests. **Your PR cannot be merged unless tests pass**_

### CODE REVIEW

File: tasmota/tasmota_template.h [hunk 1]
```diff
@@ -101,6 +101,7 @@ enum UserSelectablePins {
   GPIO_ADC_BUTTON, GPIO_ADC_BUTTON_INV,  // Analog Button
   GPIO_ADC_RANGE,                      // Analog Range
   GPIO_ADC_CT_POWER,                   // ANalog Current
+  GPIO_ADC_PH,                         // Analog PH Sensor
```
Comment (arendst) [inline tasmota/tasmota_template.h:104]: Inserting it here will break every current installation. Needs to go at the end of the list.

File: tasmota/tasmota_template.h [hunk 2]
```diff
@@ -266,6 +267,7 @@ const char kSensorNames[] PROGMEM =
   D_SENSOR_ADC_BUTTON "|" D_SENSOR_ADC_BUTTON "_i|"
   D_SENSOR_ADC_RANGE "|"
   D_SENSOR_ADC_CT_POWER "|"
+  D_SENSOR_ADC_PH "|"
```
Comment (arendst) [inline tasmota/tasmota_template.h:270]: Ditto

File: tasmota/tasmota_template.h [hunk 3]
```diff
@@ -761,6 +763,7 @@ const uint16_t kGpioNiceList[] PROGMEM = {
   AGPIO(GPIO_ADC_RANGE) + MAX_ADCS,       // Range
   AGPIO(GPIO_ADC_CT_POWER) + MAX_ADCS,    // Current
   AGPIO(GPIO_ADC_JOY) + MAX_ADCS,         // Joystick
+  AGPIO(GPIO_ADC_PH) + MAX_ADCS,          // Analog PH Sensor
```
Comment (arendst) [inline tasmota/tasmota_template.h:766]: OK

File: tasmota/tasmota_template.h [hunk 4]
```diff
@@ -779,6 +782,7 @@ const uint16_t kAdcNiceList[] PROGMEM = {
   AGPIO(GPIO_ADC_RANGE),                  // Range
   AGPIO(GPIO_ADC_CT_POWER),               // Current
   AGPIO(GPIO_ADC_JOY),                    // Joystick
+  AGPIO(GPIO_ADC_PH),                     // Analog PH Sensor
```
Comment (arendst) [inline tasmota/tasmota_template.h:785]: OK

File: tasmota/language/af_AF.h [hunk 5]
```diff
@@ -740,6 +740,7 @@
 #define D_SENSOR_ADC_RANGE     "ADC Reekse"
 #define D_SENSOR_ADC_CT_POWER  "ADC CT Power"
 #define D_SENSOR_ADC_JOYSTICK  "ADC Joystick"
+#define D_SENSOR_ADC_PH        "Analoog pH Sensor"
```
Comment (arendst) [inline tasmota/language/af_AF.h:743]: Change to ``ADC pH`` only. As short as possible!!!

For all languages pls.

**Comment by SimonFili** [comment]
Hi there,  Trying to use this adcparam 9, but it's not working.  I think it's because my linear (slope) curve is negative.
Here's my raw values and pH reading 

RAW	pH
601	        12,7
2644	3,12

And the parameter I enter:
adcparam1 9,3.12,2644,12.7,601

So ADC goes up when pH goes down.

This is with an ESP32 with Tastmota 10.1.0

**Comment by barbudor** [comment]
@SimonFili it was fixed with negative slope in  https://github.com/arendst/Tasmota/issues/14245
Please provide logs with Weblog 3

**Comment by SimonFili** [comment]
Here you go: Lots of sensors on this device. Let me know if you are looking for something specific.

05:49:19.023 CMD: weblog 3
05:49:19.025 SRC: WebConsole from 192.168.10.147
05:49:19.027 CMD: Grp 0, Cmnd 'WEBLOG', Idx 1, Len 1, Data '3'
05:49:19.043 MQT: stat/T3ED04/RESULT = {"WebLog":3}
05:49:21.432 APP: Analog pH read. ADC-RAW: 2050, cal-low(pH=ADC): 3.11=2644, cal-high(pH=ADC): 12.69=601
05:49:24.362 APP: Analog pH read. ADC-RAW: 2027, cal-low(pH=ADC): 3.11=2644, cal-high(pH=ADC): 12.69=601
05:49:26.437 BRY: GC from 4660 to 3710 bytes, objects freed 2/48 (in 1 ms)
05:49:26.451 MQT: tele/T3ED04/STATE = {"Time":"2022-02-24T05:49:26","Uptime":"0T09:06:54","UptimeSec":32814,"Heap":157,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":20,"MqttCount":14,"Berry":{"HeapUsed":3,"Objects":48},"Wifi":{"AP":1,"SSId":"CirculusGrow","BSSId":"A6:DA:C4:A4:46:F2","Channel":11,"Mode":"11n","RSSI":72,"Signal":-64,"LinkCount":15,"Downtime":"0T00:07:48"}}
05:49:26.466 APP: Analog pH read. ADC-RAW: 2048, cal-low(pH=ADC): 3.11=2644, cal-high(pH=ADC): 12.69=601
05:49:26.490 MQT: tele/T3ED04/SENSOR = {"Time":"2022-02-24T05:49:26","Switch1":"ON","Switch2":"ON","COUNTER":{"C1":10134,"C2":9719},"ANALOG":{"pH1":12.69},"BME680-76":{"Temperature":14.4,"Humidity":27.8,"DewPoint":-4.0,"Pressure":1003.4,"Gas":342.50},"BME680-77":{"Temperature":14.5,"Humidity":28.0,"DewPoint":-3.9,"Pressure":1003.3,"Gas":368.81},"MHZ19B":{"Model":"","CarbonDioxide":0,"Temperature":0.0},"ESP32":{"Temperature":41.7},"PressureUnit":"hPa","TempUnit":"C"}
05:49:27.425 MQT: tele/T3ED04/HASS_STATE = {"Version":"10.1.0(tasmota)","BuildDateTime":"2021-12-08T14:50:20","Module or Template":"ESP32-DevKit","RestartReason":"Software reset CPU","Uptime":"0T09:06:55","Hostname":"T3ED04-1080","IPAddress":"192.168.10.129","RSSI":"70","Signal (dBm)":"-65","WiFi LinkCount":15,"WiFi Downtime":"0T00:07:48","MqttCount":14,"LoadAvg":22}
05:49:27.861 APP: Analog pH read. ADC-RAW: 2032, cal-low(pH=ADC): 3.11=2644, cal-high(pH=ADC): 12.69=601
05:49:30.423 APP: Analog pH read. ADC-RAW: 2026, cal-low(pH=ADC): 3.11=2644, cal-high(pH=ADC): 12.69=601
05:49:33.375 APP: Analog pH read. ADC-RAW: 2036, cal-low(pH=ADC): 3.11=2644, cal-high(pH=ADC): 12.69=601
05:49:36.373 APP: Analog pH read. ADC-RAW: 2042, cal-low(pH=ADC): 3.11=2644, cal-high(pH=ADC): 12.69=601

**Comment by SimonFili** [comment]
image
This is what I'm trying to fit

**Comment by barbudor** [comment]
Just to be sure, are running recent version? At least 11.0? Or current dev?

**Comment by barbudor** [comment]
No, you're not!

**Comment by SimonFili** [comment]
I'm with 10.1, doing a firmware update now.  I did not know 11 was out.

**Comment by SimonFili** [comment]
Wow!  Works ok with V 11.0! Thanks for this.  So there was a bug in < v11.

Thanks again!  Made my day ;-)

**Pull request merged**